prov/gni/test/fabric.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2015-2016 Cray Inc. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <getopt.h>
+#include <poll.h>
+#include <time.h>
+#include <string.h>
+
+
+#include "gnix.h"
+
+#include <criterion/criterion.h>
+#include "gnix_rdma_headers.h"
+#include "fi_ext_gni.h"
+
+static struct fid_fabric *fabric;
+static struct fi_info *fi;
+
+static void setup(void)
+{
+	int ret;
+	struct fi_info *hints;
+
+	hints = fi_allocinfo();
+	cr_assert(hints, "fi_allocinfo");
+
+	hints->fabric_attr->prov_name = strdup("gni");
+
+	ret = fi_getinfo(FI_VERSION(1, 0), NULL, 0, 0, hints, &fi);
+	cr_assert(ret == FI_SUCCESS, "fi_getinfo");
+
+	ret = fi_fabric(fi->fabric_attr, &fabric, NULL);
+	cr_assert(ret == FI_SUCCESS, "fi_fabric");
+
+	fi_freeinfo(hints);
+}
+
+static void teardown(void)
+{
+	int ret;
+
+	ret = fi_close(&fabric->fid);
+	cr_assert(ret == FI_SUCCESS, "fi_close fabric");
+
+	fi_freeinfo(fi);
+}
+
+TestSuite(fabric, .init = setup, .fini = teardown);
+
+
+
+
+Test(fabric, simple) 
+{
+	cr_assert(fabric != NULL);
+}
+
+Test(fabric, open_ops_1)
+{
+	int ret;
+	struct fi_gni_ops_fab *ops; 
+
+	ret = fi_open_ops(&fabric->fid, FI_GNI_FAB_OPS_1, 0, (void **) &ops, NULL);
+	cr_assert_eq(ret, FI_SUCCESS);
```

Comment:
I found it useful to use: `cr_assert(ret == FI_SUCCESS, "fi_open_ops returned: %s", fi_strerror(-ret));` while testing the FI_ADDR_STR changes.